string map_file_path
---
bool success
